### PR TITLE
CHAIN-559: persist BTC network fee for withdrawals

### PR DESCRIPTION
**Pull request by bflood366**

### CODE REVIEW

File: backend/src/main/kotlin/xyz/funkybit/core/utils/bitcoin/ArchUtils.kt [hunk 1]
```diff
@@ -453,17 +453,26 @@ object ArchUtils {
...
+                    TransactionOutPoint(
+                        params,
+                        it.vout(),
+                        Sha256Hash.wrap(it.txId().value),
+                    ),
+                    Coin.valueOf(it.amount.toLong()),
+                ).also { input ->
+                    input.setScriptSig(ScriptBuilder.createEmpty())
+                    // this matches the size of the elements pushed onto witness stack when arch signs an input
+                    input.witness = TransactionWitness(3).also { transactionWitness ->
```
Comment (bflood366) [inline backend/src/main/kotlin/xyz/funkybit/core/utils/bitcoin/ArchUtils.kt:470]: the way the fee is calculated is we get the recommended fee and then build a tx to emulate what the program will send and multiply the tx vsize by recommended fee (sats/vbyte). We then use that given the input and output utxos to determine the change amount.  I noticed that the vsize of the actual tx sent on chain and our estimate here was different (so the effective sats/vbyte we were paying was lower than the recommended fee). The issue was we using P2WPKH inputs in the estimate, but the inputs are taproot - so this builds a witness stack with same length items as the program builds for the taproot inputs.

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/BlockchainTransaction.kt [hunk 2]
```diff
@@ -82,6 +83,13 @@ class BlockchainTransactionEntity(guid: EntityID<BlockchainTransactionId>) : GUI
             }
             return entity
         }
+
+        fun findByChainAndHash(chainId: ChainId, txHash: TxHash): BlockchainTransactionEntity? {
+            return BlockchainTransactionEntity.find {
+                BlockchainTransactionTable.chainId.eq(chainId) and
```
Comment (bholzman) [inline backend/src/main/kotlin/xyz/funkybit/core/model/db/BlockchainTransaction.kt:89]: noticed we don't have an index on either txHash or chainId (probably just on txHash is sufficient)

**Review by bholzman — APPROVED** [review]
Looks good!

File: backend/src/main/kotlin/xyz/funkybit/core/model/db/BlockchainTransaction.kt (hunk 2 above)
Comment (bflood366) [reply backend/src/main/kotlin/xyz/funkybit/core/model/db/BlockchainTransaction.kt:89]: added an index on tx_hash

**Pull request merged**